### PR TITLE
Igorjuraszek/sc 34/add contributions new route with all form

### PR DESCRIPTION
**Pull request by igorjuraszek**

### CODE REVIEW

File: app/components/new-contribution/contributors/row/template.hbs [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  <td>{{@contributionUser.amount}} PLN</td>
+  <td></td>
+  <td><button
+      type='button'
+      class='btn btn-link card-link'
+      {{on 'click' this.onDeleteContributor}}
+    >Delete</button></td>
+</tr>
+
+{{yield}}
```
Comment (wozny1989) [inline app/components/new-contribution/contributors/row/template.hbs:12]: Remove if you are not using this component blocky ✂️

File: app/components/new-contribution/contributors/template.hbs [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{{#each @contribution.contributors as |contribution-user|}}
+  <NewContribution::Contributors::Row @contributionUser={{contribution-user}} />
+{{/each}}
+{{yield}}
```
Comment (wozny1989) [inline app/components/new-contribution/contributors/template.hbs:4]: Remove if you are not using this component blocky ✂️

File: app/home/contribution/new/controller.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  @tracked amount = null;
+
+  get currentlyAddedContributors() {
+    const currentlyAddedContributors = this.model.contributors.map(
+      (contributionRow) => {
+        return contributionRow.contributor.get('id');
+      }
+    );
+    return currentlyAddedContributors;
+  }
```
Comment (wozny1989) [inline app/home/contribution/new/controller.js:21]:
```suggestion
  get currentlyAddedContributors() {
    return this.model.contributors.map(({ contributor }) => contributor.get('id'));
  }
```

File: app/home/contribution/new/controller.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    return this.session.currentUser;
+  }
+  get currentContribution() {
+    return this.model;
+  }
+
+  get goalOfContribution() {
+    const contributorsDebts = this.model.contributors.map((contributor) => {
+      return parseFloat(contributor.amount);
+    });
```
Comment (wozny1989) [inline app/home/contribution/new/controller.js:37]:
```suggestion
    const contributorsDebts = this.model.contributors.map(({ amount }) => parseFloat(amount));
```

File: app/home/contribution/new/template.hbs [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+        />
+
+        <h3>Owner: {{this.contributionOwner.username}}</h3>
+        <h4>Deadline: -</h4>
+
+        <div class='form-check'>
+          <input
+            class='form-check-input'
+            type='checkbox'
+            value=''
```
Comment (wozny1989) [inline app/home/contribution/new/template.hbs:23]: ✂️

File: app/home/contribution/new/template.hbs [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+            id='checkIsPrivate'
+            {{on 'change' this.onChangePrivate}}
+          />
+          <label class='form-check-label' for='flexCheckDefault'>
+            Private contribution
+          </label>
+        </div>
+      </div>
+
+      {{! template-lint-disable table-groups }}
```
Comment (wozny1989) [inline app/home/contribution/new/template.hbs:33]: ✂️

File: app/home/contributions/show/route.js [hunk 7]
```diff
@@ -0,0 +1,13 @@
+import Route from '@ember/routing/route';
+import { inject as service } from '@ember/service';
+
+export default class HomeContributionsShowRoute extends Route {
+  @service store;
+
+  model() {
+    // const contributionUser = this.store.findAll('contribution-user');
+    // return contributionUser;
```
Comment (wozny1989) [inline app/home/contributions/show/route.js:9]: ✂️

File: app/home/contribution/new/template.hbs [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+        type='submit'
+        class='btn btn-primary'
+        {{on 'click' this.onSubmit}}
+      >Create</button>
+    </div>
+  </form>
+
+</div>
+
+{{outlet}}
```
Comment (wozny1989) [inline app/home/contribution/new/template.hbs:100]: Remove `{{outlet}}` if you not have any nested routes

**Review by wozny1989 — CHANGES_REQUESTED** [review]

File: app/home/contributions/show/template.hbs [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+    </div>
+    <div class='card-footer text-muted'>
+      2 days remaining
+    </div>
+  </div>
+</div>
+
+{{outlet}}
+
+{{! https://getbootstrap.com/docs/4.0/content/tables/ }}
```
Comment (Rxbsxn) [inline app/home/contributions/show/template.hbs:116]: ✂️

File: app/home/contributions/show/template.hbs [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+
+      <a href='#'>details</a>
+    </div>
+    <div class='card-footer text-muted'>
+      2 days remaining
+    </div>
+  </div>
+</div>
+
+{{outlet}}
```
Comment (Rxbsxn) [inline app/home/contributions/show/template.hbs:114]: ✂️

**Pull request merged**